kasa/device.py

```diff
@@ -248,15 +250,15 @@ def is_color(self) -> bool:
         """Return True if the device supports color changes."""
         return False
 
-    def get_plug_by_name(self, name: str) -> "Device":
+    def get_plug_by_name(self, name: str) -> Device:
```

Comment:
```suggestion
    def get_plug_by_name(self, name: str) -> Self:
```
I'm wondering if [the self type](https://peps.python.org/pep-0673/) would also work?

Reply:
Nah, I don't think we want new imports, but I thought it would "just work" considering the futures import defers the interpretation. Could you try to change it for testing to see if the CI passes?

Reply:
Actually I'm not sure if `Self` is correct in this context.  It's a generic type so implies that the child plug is the same type as the parent. I'm not sure if that's correct for the IOT devices when they get their children.  It's also not a change related to this PR really so I think we should leave as-is for now.

Reply:
`ruff check`
`kasa/device.py:253:46: F821 Undefined name `Self``

It needs to be imported from the `typing` and it's not there until [python3.11](https://peps.python.org/pep-0673/)

Reply:
Ah, okay, let's skip it then. And you are right, it wouldn't be valid in this context anyway..